### PR TITLE
The Tramstation Tram no longer shoves the handicapped out of their wheelchairs when moved

### LINKED ISSUE #60375: Riding on tram as a paraplegic will cause you to be slammed by it.
## Reproduction:
Be paraplegic. Go onto tram. Once done, you will fall off your wheelchair (this happened before), be hit by the tram, and then have face/eyewear fall through the tram itself.

### PR DESCRIPTION
**Pull request by MMMiracles**
## About The Pull Request

Adds a mob trait that is used to prevent forcemoves from unbuckling mobs, which in turn fixes the tram's issue of unbuckling you from your seats when it moves. This trait is added when you are added to the list of items to move and you lose the trait as soon as you leave the tram itself, hopefully meaning this shouldn't have any unforeseen consequences. 

## Why It's Good For The Game

You can now buckle yourself into the seats on the tram and actually be in the chair. The crippled crew will no longer be bullied by the tram by being forced to crawl back into their wheelchair after departing.

## Changelog

:cl: MMMiracles
fix: The Tramstation Tram no longer force unbuckles you from chairs/vehicles when moving.
/:cl:

Fixes #60375

I've played around a bit locally to make sure you can't escape the tram with the buckle trait. The only issue I'm running into is that when you're mounted on rideables like the wheelchair, the seat lags behind due to what I assume is an issue with gliding or relaymove. The slower the vehicle, the worse the catchup (it still lets you move properly while mid-transit so it seems like mostly just a visual issue atm)

### CODE REVIEW

File: code/_globalvars/traits.dm [hunk 1]
```diff
@@ -168,6 +168,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_UNNATURAL_RED_GLOWY_EYES" = TRAIT_UNNATURAL_RED_GLOWY_EYES,
 		"TRAIT_BLOODSHOT_EYES" = TRAIT_BLOODSHOT_EYES,
 		"TRAIT_SHIFTY_EYES" = TRAIT_SHIFTY_EYES,
+		"TRAIT_CANNOT_BE_UNBUCKLED" = TRAIT_CANNOT_BE_UNBUCKLED
```
Comment (Mothblocks) [inline code/_globalvars/traits.dm:171]:
```suggestion
		"TRAIT_CANNOT_BE_UNBUCKLED" = TRAIT_CANNOT_BE_UNBUCKLED,
```

File: code/game/objects/structures/industrial_lift.dm [hunk 2]
```diff
@@ -189,6 +192,9 @@ GLOBAL_LIST_EMPTY(lifts)
 		return
 	if(AM in lift_load)
 		return
+	if(isliving(AM) && !HAS_TRAIT(AM, TRAIT_CANNOT_BE_UNBUCKLED))
+		var/mob/living/hewhocannotbeunbuckled = AM
```
Comment (Mothblocks) [inline code/game/objects/structures/industrial_lift.dm:196]: You don't need this variable

File: code/game/objects/structures/industrial_lift.dm [hunk 3]
```diff
@@ -180,6 +180,9 @@ GLOBAL_LIST_EMPTY(lifts)
 	SIGNAL_HANDLER
 	if(!(potential_rider in lift_load))
 		return
+	if(isliving(potential_rider) && HAS_TRAIT(potential_rider, TRAIT_CANNOT_BE_UNBUCKLED))
+		var/mob/living/hewhocanbeunbuckled = potential_rider
```
Comment (Mothblocks) [inline code/game/objects/structures/industrial_lift.dm:184]: You don't need this variable

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
> The only issue I'm running into is that when you're mounted on rideables like the wheelchair, the seat lags behind due to what I assume is an issue with gliding or relaymove. The slower the vehicle, the worse the catchup.

It most surely has something to do with movables that have a riding component having their glide size set by said component any time they are moved and not only when they are driven/ridden. I'm on it.

**Comment by notfrying1pans** [comment]
blessed

**Review by Mothblocks — APPROVED** [review]

**Comment by GoldenAlpharex** [comment]
@MMMiracles Could you test just to be sure if this fixes the elevators going up and down crushing those poor people on wheelchairs? I feel like it will, but you can never be too sure.

**Comment by Ghommie** [comment]
@81Denton I'd be glad if you could merge #63048 soon too so I don't have to edit a cl entry in that says vehicles won't lag behind inside trams anymore or something.

**Pull request merged**